### PR TITLE
Add customer company enrichment

### PR DESCRIPTION
**Pull request by justnero**
## Summary
Adds a customer company enrichment

## Checklist

- [x] Writing style
- [x] API design standards

### CODE REVIEW

**Review by ashkarpetin — DISMISSED** [review]

File: openapi/components/schemas/Company.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+type: object
+description: |-
+  Company information associated with the customer primary email address domain.
```
Comment (noon-dawg) [inline openapi/components/schemas/Company.yaml:3]: nitpick: ( we can use [possessive apostrophes](https://github.com/Rebilly/api-definitions/blob/main/WRITING-STYLE.md#possessive-apostrophes) for people)
```suggestion
  Company information that is associated with the customer's primary email address domain.
```

File: openapi/components/schemas/Company.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  
+  This is a paid feature, to enable it [contact Rebilly](https://www.rebilly.com/support/).
+nullable: true
+readOnly: true
+properties:
+  name:
+    type: string
```
Comment (noon-dawg) [inline openapi/components/schemas/Company.yaml:10]:
```suggestion
  name:
    type: string
    description: Name of the company.
```

File: openapi/components/schemas/Company.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  domain:
+    type: string
+    nullable: true
+  yearFounded:
+    type: number
+    format: integer
+    nullable: true
+  industry:
+    type: string
+    nullable: true
```
Comment (noon-dawg) [inline openapi/components/schemas/Company.yaml:21]:
```suggestion
    type: string
    description: Industry the company is associated with.
    nullable: true
```

File: openapi/components/schemas/Company.yaml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  employeesCount:
+    type: number
+    format: integer
```
Comment (noon-dawg) [inline openapi/components/schemas/Company.yaml:24]:
```suggestion
    type: number
    description: Number of employees in the company.
    format: integer
```

File: openapi/components/schemas/Company.yaml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    nullable: true
+  country:
+    type: string
+    nullable: true
```
Comment (noon-dawg) [inline openapi/components/schemas/Company.yaml:28]:
```suggestion
    type: string
    description: Country where the company is based.
    nullable: true
```

File: openapi/components/schemas/Company.yaml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  locality:
+    type: string
+    nullable: true
+  linkedinUrl:
+    type: string
+    format: url
```
Comment (noon-dawg) [inline openapi/components/schemas/Company.yaml:34]: suggestion: I'm not sure if this is correct
```suggestion
    type: string
    description: URL of the company website.
    format: url
```

File: openapi/components/schemas/Company.yaml [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+  
+  This is a paid feature, to enable it [contact Rebilly](https://www.rebilly.com/support/).
+nullable: true
+readOnly: true
+properties:
+  name:
+    type: string
+    nullable: true
+  domain:
+    type: string
```
Comment (noon-dawg) [inline openapi/components/schemas/Company.yaml:13]: Issue: We need a description for this field. (I'm not sure what domain represents)

**Review by noon-dawg — CHANGES_REQUESTED** [review]
We should add descriptions for all properties.

I've made at attempt at adding them here, I'm not sure if they are correct 😅

File: openapi/components/schemas/Company.yaml [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+      properties:
+        href:
+          description: Link URL.
+          type: string
+        rel:
+          description: Type of link.
+          type: string
+          enum:
+            - linkedin
+
```
Comment (justnero) [inline openapi/components/schemas/Company.yaml:45]:
```suggestion

```

File: openapi/components/schemas/Company.yaml [hunk 9]
```diff
@@ -1,33 +1,39 @@
...
   name:
     type: string
+    description: Name of the company.
     nullable: true
   domain:
     type: string
     nullable: true
   yearFounded:
     type: number
+    description: Year the company was founded.
```
Comment (noon-dawg) [inline openapi/components/schemas/Company.yaml:18]: suggestion: fixes the lint issue
```suggestion
    type: number
    description: Founding year of the company.
```

File: openapi/components/schemas/Company.yaml [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+  This is a paid feature, to enable it [contact Rebilly](https://www.rebilly.com/support/).
+nullable: true
+readOnly: true
+properties:
+  name:
+    type: string
+    description: Name of the company.
+    nullable: true
+  domain:
+    type: string
```
Comment (noon-dawg) [inline openapi/components/schemas/Company.yaml:14]: Issue: We need a description for this field. (I'm not sure if my suggestion is correct)
```suggestion
    type: string
    description: Website domain of the company.
```

**Review by noon-dawg — DISMISSED** [review]
The descriptions look good to me!

**Review by noon-dawg — DISMISSED** [review]

**Review by ashkarpetin — DISMISSED** [review]

File: openapi/components/schemas/Company.yaml [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+    description: Website domain of the company.
+    nullable: true
```
[the remark this answers is not in the document]
Comment (justnero) [reply openapi/components/schemas/Company.yaml:16]: Nice catch, thank you!
```suggestion
```

**Review by noon-dawg — APPROVED** [review]

**Review by vykhrystiuk — APPROVED** [review]

**Review by ashkarpetin — APPROVED** [review]

**Pull request merged**